core/types/subtyping.cc

```diff
@@ -1383,9 +1383,21 @@ bool isSubTypeUnderConstraintSingle(const GlobalState &gs, TypeConstraint &const
                                 break;
                             }
                         }
-                        auto message = ErrorColors::format("`{}` is not {} `{}` for {} type member `{}`", a1i.show(gs),
-                                                           joiningText, a2j.show(gs), variance, idxTypeMember.show(gs));
-                        subCollector.message = message;
+                        // For proc types, make type member references more user-friendly
+                        if (a2->klass.data(gs)->name.show(gs).compare("Proc1") == 0) {
+                            auto message = ErrorColors::format("`{}` is not {} `{}`", a1i.show(gs), joiningText, a2j.show(gs));
+                            if (variance == "covariant") {
+                                subCollector.message = fmt::format("{} for return type of Proc1", message);
```

Comment:
This should say something like "return type of T.proc" or "argument N type of T.proc". Continuing to show `Arg0` and `Proc1` does not fix the original issue, which is that we're trying to hide these classes from the user.